.github/workflows/ci.yml

```diff
@@ -34,119 +34,86 @@ jobs:
             PATHFINDER_ETHEREUM_HTTP_MAINNET_URL=${{ secrets.ALCHEMY_MAINNET_URL }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_PASSWORD=${{ secrets.ALCHEMY_MAINNET_PASSWORD }} \
-            timeout 3m ./pathfinder_test ethereum::
+            timeout 3m ./pathfinder_ethereum
 
-  test_infura:
-    name: pathfinder tests (infura)
-    needs: compile_tests
+  test_pathfinder_ethereum_infura:
+    name: pathfinder-ethereum tests (infura)
+    needs: compile_pathfinder_ethereum_tests
     runs-on: ubuntu-latest
     steps:       
       # Fetch the test binary artifact
       - uses: actions/download-artifact@v3
         with:
-          name: test-binary
+          name: test-binary-pathfinder-ethereum
       # Archiving removes the file permissions
       - name: Fix file permissions
-        run: chmod +x pathfinder_test
+        run: chmod +x pathfinder_ethereum
 
       - name: Execute tests
         run: |
           PATHFINDER_ETHEREUM_HTTP_GOERLI_URL=${{ secrets.INFURA_GOERLI_URL }} \
             PATHFINDER_ETHEREUM_HTTP_GOERLI_PASSWORD=${{ secrets.INFURA_GOERLI_PASSWORD }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_URL=${{ secrets.INFURA_MAINNET_URL }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_PASSWORD=${{ secrets.INFURA_MAINNET_PASSWORD }} \
-            timeout 3m ./pathfinder_test ethereum::
+            timeout 3m ./pathfinder_ethereum
 
-  test_stark_curve:
-    name: stark_curve tests
+  test_all_except_pathfinder_ethereum:
+    name: all tests w/o pathfinder-ethereum
     runs-on: ubuntu-latest
     steps:
+      # Required for the python environment
       - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - name: tests
-        run: cargo test -p stark_curve --all-targets --locked
 
-  test_stark_hash:
-    name: stark_hash tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - name: tests
-        run: cargo test -p stark_hash --all-targets --locked
+      - uses: actions/setup-python@v4
+        id: setup_python
+        with:
+          python-version: '3.8'
+          cache: 'pip'
+          cache-dependency-path: 'py/requirements*.txt'
+      # Workaround for setup-python issue #330 (https://github.com/actions/setup-python/issues/330)
+      - uses: actions/cache@v3
+        id: python_cache
+        with:
+          path: ${{ env.pythonLocation }}/**/site-packages
+          key: site-packages-${{ steps.setup_python.outputs.python-version }}-${{ hashFiles('py/requirements*.txt') }}
+      - run: pip install -e py/.
+
+      # We want to strip the debug info as some of these test binaries are 100MB+ and pathfinder_lib is 1GB+
+      - name: Execute tests
+        run: cargo test --all-targets --workspace --locked --profile stripdebuginfo --exclude pathfinder-ethereum
```

Comment:
And didn't we have a `timeout` originally for running tests?

Reply:
15m? wdyt? the last run took 11m+